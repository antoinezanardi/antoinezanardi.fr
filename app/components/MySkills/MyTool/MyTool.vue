<template>
  <WrappedFontAwesomeIcon
    class="mx-1 tool"
    data-bs-toggle="tooltip"
    :icon="tool.iconClasses"
    :icon-color="tool.color"
    role="alertdialog"
    size="3x"
    :title="tool.description"
  />
</template>

<script setup lang="ts">
import type { MyToolProps } from "~/components/MySkills/MyTool/my-tool.types";
import WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";

defineProps<MyToolProps>();
</script>

<style lang="scss" scoped>
  .tool {
    cursor: pointer;
  }
</style>