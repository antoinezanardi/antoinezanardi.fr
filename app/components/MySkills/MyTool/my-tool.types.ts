import type { Tool } from "~/models/tool/tool.types";

type MyToolProps = {
  tool: Tool;
};

export type { MyToolProps };