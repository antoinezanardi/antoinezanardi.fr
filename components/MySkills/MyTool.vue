<template>
  <i
    class="tool fa-3x mx-1"
    :class="tool.iconClasses"
    data-bs-toggle="tooltip"
    :title="tool.description"
  />
</template>

<script setup lang="ts">
import type { Tool } from "~/models/Tool";

defineProps<{
  tool: Tool;
}>();
</script>

<style lang="scss" scoped>
  .tool {
    color: v-bind("tool.color");
    cursor: pointer;
  }
</style>