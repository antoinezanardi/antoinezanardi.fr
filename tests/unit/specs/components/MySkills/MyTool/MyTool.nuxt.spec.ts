import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { MyToolProps } from "~/components/MySkills/MyTool/my-tool.types";
import MyTool from "~/components/MySkills/MyTool/MyTool.vue";
import type WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";

describe("My Tool Component", () => {
  const defaultProps: MyToolProps = {
    tool: {
      color: "#3498db",
      description: "MySkills.tools.vuejs",
      iconClasses: "fab fa-vuejs",
    },
  };
  let wrapper: ReturnType<typeof mount<typeof MyTool>>;

  async function mountMyToolComponent(options: ComponentMountingOptions<typeof MyTool> = {}): Promise<ReturnType<typeof mount<typeof MyTool>>> {
    return mountSuspendedComponent(MyTool, {
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountMyToolComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Tool Icon", () => {
    it("should pass icon classes when rendered.", () => {
      const toolIcon = wrapper.findComponent<typeof WrappedFontAwesomeIcon>(".tool");

      expect(toolIcon.attributes("icon")).toBe(defaultProps.tool.iconClasses);
    });

    it("should pass tool color when rendered.", () => {
      const toolIcon = wrapper.findComponent<typeof WrappedFontAwesomeIcon>(".tool");

      expect(toolIcon.attributes("iconcolor")).toBe(defaultProps.tool.color);
    });

    it("should pass tool description when rendered.", () => {
      const toolIcon = wrapper.findComponent<typeof WrappedFontAwesomeIcon>(".tool");

      expect(toolIcon.attributes("title")).toBe(defaultProps.tool.description);
    });
  });
});