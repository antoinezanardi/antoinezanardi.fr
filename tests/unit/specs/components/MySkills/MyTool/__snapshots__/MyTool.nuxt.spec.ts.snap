// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Tool Component > should match snapshot when rendered. 1`] = `"<wrapped-font-awesome-icon-stub data-v-c01e7f82="" icon="fab fa-vuejs" iconcolor="#3498db" size="3x" classes="" class="mx-1 tool" data-bs-toggle="tooltip" role="alertdialog" title="MySkills.tools.vuejs"></wrapped-font-awesome-icon-stub>"`;
